packages/serverpod/lib/src/endpoints/insights.dart

```diff
@@ -319,4 +320,27 @@ class InsightsEndpoint extends Endpoint {
       );
     }
   }
+
+  /// Fetches a file from the server. Only whitelisted files can be fetched.
+  /// The file path must be in unix format and relative to the servers root
+  /// directory.
+  Future<String> fetchFile(Session session, String path) async {
```

Comment:
Question: For my knowledge - Is there any authentication protecting this endpoint?

My concern is that this would otherwise expose all protocol files for all serverpod servers.